stats/redis.go

```diff
@@ -0,0 +1,118 @@
+// Licensed under the Apache License, Version 2.0
+// Details: https://raw.githubusercontent.com/maniksurtani/quotaservice/master/LICENSE
+
+package stats
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/maniksurtani/quotaservice/events"
+	"github.com/maniksurtani/quotaservice/logging"
+
+	"gopkg.in/redis.v3"
+)
+
+type redisListener struct {
+	client *redis.Client
+}
+
+func NewRedisStatsListener(redisOpts *redis.Options) *redisListener {
+	client := redis.NewClient(redisOpts)
```

Comment:
Can we share the redis client with buckets/redis? Maybe abstract some of the connection management stuff?